File: config/src/main/java/org/wcl/mfn/config/ui/tools/ContractCalculatorConfig.java
package org.wcl.mfn.config.ui.tools;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.PropertySource;

@Configuration
@PropertySource("ui/tools/contract-calculator.properties")
public class ContractCalculatorConfig {
    @Value("${contract-calculator.view.attribute.contract_parameters}")
    private String contractParameters;
    @Value("${contract-calculator.view.attribute.suggested_contracts}")
    private String suggestedContracts;

    @Value("${mfn.contract-calculator.salary-label}")
    private String salaryLabel;
    @Value("${mfn.contract-calculator.bonus-label}")
    private String bonusLabel;
    @Value("${mfn.contract-calculator.escalator-label}")
    private String escalatorLabel;
    @Value("${mfn.contract-calculator.escalator}")
    private String escalator;
    @Value("${mfn.contract-calculator.escalator-value}")
    private String escalatorValue;
    @Value("${mfn.contract-calculator.reset-button}")
    private String resetButton;
    @Value("${mfn.contract-calculator.submit-button}")
    private String submitButton;
    @Value("${mfn.contract-calculator.suggested-contracts.table}")
    private String suggestedContractsTable;
    @Value("${mfn.contract-calculator.suggested-contracts.main-header}")
    private String suggestedContractsMainHeader;
    @Value("${mfn.contract-calculator.suggested-contracts.sub-header.years}")
    private String suggestedContractsSubHeaderYears;
    @Value("${mfn.contract-calculator.suggested-contracts.sub-header.salary}")
    private String suggestedContractsSubHeaderSalary;
    @Value("${mfn.contract-calculator.suggested-contracts.sub-header.remuneration}")
    private String suggestedContractsSubHeaderRemuneration;
    @Value("${mfn.contract-calculator.suggested-contracts.sub-header.contract-year}")
    private String suggestedContractsSubHeaderContractYear;
    @Value("${mfn.contract-calculator.suggested-contracts.sub-header.contract-year-salary}")
    private String suggestedContractsSubHeaderContractYearSalary;
    @Value("${mfn.contract-calculator.suggested-contracts.sub-header.contract-year-bonus}")
    private String suggestedContractsSubHeaderContractYearlyBonus;
    @Value("${mfn.contract-calculator.suggested-contracts.sub-header.contract-year-total}")
    private String suggestedContractsSubHeaderContractYearlyTotal;
    @Value("${mfn.contract-calculator.suggested-contracts.contract-years}")
    private String suggestedContractsContractYears;
    @Value("${mfn.contract-calculator.suggested-contracts.total-salary}")
    private String suggestedContractsTotalSalary;
    @Value("${mfn.contract-calculator.suggested-contracts.total-remuneration}")
    private String suggestedContractsTotalRemuneration;
    @Value("${mfn.contract-calculator.suggested-contracts.current-year}")
    private String suggestedContractsCurrentYear;
    @Value("${mfn.contract-calculator.suggested-contracts.year-salary}")
    private String suggestedContractsYearSalary;
    @Value("${mfn.contract-calculator.suggested-contracts.bonus-per-year}")
    private String suggestedContractsBonusPerYear;
    @Value("${mfn.contract-calculator.suggested-contracts.total-per-year}")
    private String suggestedContractsTotalPerYear;

    public String contractParameters() { return contractParameters; }
    public String suggestedContracts() { return suggestedContracts;}

    public String salaryLabel() { return salaryLabel; }
    public String bonusLabel() { return bonusLabel; }
    public String escalatorLabel() { return escalatorLabel; }
    public String escalator() { return escalator; }
    public String escalatorValue() { return escalatorValue; }
    public String resetButton() { return resetButton; }
    public String submitButton() { return submitButton; }

    public String suggestedContractsTable() { return suggestedContractsTable;}
    public String suggestedContractsMainHeader() { return suggestedContractsMainHeader;}
    public String suggestedContractsSubHeaderYears() { return suggestedContractsSubHeaderYears;}
    public String suggestedContractsSubHeaderSalary() { return suggestedContractsSubHeaderSalary;}
    public String suggestedContractsSubHeaderRemuneration() { return suggestedContractsSubHeaderRemuneration;}
    public String suggestedContractsSubHeaderContractYear() { return suggestedContractsSubHeaderContractYear;}
    public String suggestedContractsSubHeaderContractYearSalary() { return suggestedContractsSubHeaderContractYearSalary;}
    public String suggestedContractsSubHeaderContractYearlyBonus() { return suggestedContractsSubHeaderContractYearlyBonus;}
    public String suggestedContractsSubHeaderContractYearlyTotal() { return suggestedContractsSubHeaderContractYearlyTotal;}
    public String suggestedContractsContractYears() { return suggestedContractsContractYears; }
    public String suggestedContractsTotalSalary() { return suggestedContractsTotalSalary; }
    public String suggestedContractsTotalRemuneration() { return suggestedContractsTotalRemuneration; }
    public String suggestedContractsCurrentYear() { return suggestedContractsCurrentYear; }
    public String suggestedContractsYearSalary() { return suggestedContractsYearSalary; }
    public String suggestedContractsBonusPerYear() { return suggestedContractsBonusPerYear; }
    public String suggestedContractsTotalPerYear() { return suggestedContractsTotalPerYear; }
}

File: config/src/main/kotlin/org/wcl/mfn/config/ui/tools/ContractCalculatorConfig.kt
package org.wcl.mfn.config.ui.tools

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.*

@Configuration
@PropertySource("classpath:ui/tools/contract-calculator.properties")
open class ContractCalculatorConfig {
    @Value("\${contract-calculator.view.attribute.contract_parameters}")
    private val contractParameters: String? = null

    @Value("\${contract-calculator.view.attribute.suggested_contracts}")
    private val suggestedContracts: String? = null

    @Value("\${mfn.contract-calculator.salary-label-id}")
    private val salaryLabel: String? = null

    @Value("\${mfn.contract-calculator.bonus-label-id}")
    private val bonusLabel: String? = null

    @Value("\${mfn.contract-calculator.escalator-label-id}")
    private val escalatorLabel: String? = null

    @Value("\${mfn.contract-calculator.escalator-id}")
    private val escalator: String? = null

    @Value("\${mfn.contract-calculator.escalator.default}")
    private val escalatorDefault = 0

    @Value("\${mfn.contract-calculator.escalator.max}")
    private val escalatorMax = 0

    @Value("\${mfn.contract-calculator.escalator.step}")
    private val escalatorStep = 0

    @Value("\${mfn.contract-calculator.escalator-value-id}")
    private val escalatorValue: String? = null

    @Value("\${mfn.contract-calculator.reset-button-id}")
    private val resetButton: String? = null

    @Value("\${mfn.contract-calculator.submit-button-id}")
    private val submitButton: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.table}")
    private val suggestedContractsTable: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.main-header}")
    private val suggestedContractsMainHeader: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.sub-header.years}")
    private val suggestedContractsSubHeaderYears: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.sub-header.salary}")
    private val suggestedContractsSubHeaderSalary: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.sub-header.remuneration}")
    private val suggestedContractsSubHeaderRemuneration: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.sub-header.contract-year}")
    private val suggestedContractsSubHeaderContractYear: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.sub-header.contract-year-salary}")
    private val suggestedContractsSubHeaderContractYearSalary: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.sub-header.contract-year-bonus}")
    private val suggestedContractsSubHeaderContractYearlyBonus: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.sub-header.contract-year-total}")
    private val suggestedContractsSubHeaderContractYearlyTotal: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.contract-years}")
    private val suggestedContractsContractYears: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.total-salary}")
    private val suggestedContractsTotalSalary: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.total-remuneration}")
    private val suggestedContractsTotalRemuneration: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.current-year}")
    private val suggestedContractsCurrentYear: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.year-salary}")
    private val suggestedContractsYearSalary: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.bonus-per-year}")
    private val suggestedContractsBonusPerYear: String? = null

    @Value("\${mfn.contract-calculator.suggested-contracts.total-per-year}")
    private val suggestedContractsTotalPerYear: String? = null

    fun contractParameters(): String? {
        return contractParameters
    }

    fun suggestedContracts(): String? {
        return suggestedContracts
    }

    fun salaryLabel(): String? {
        return salaryLabel
    }

    fun bonusLabel(): String? {
        return bonusLabel
    }

    fun escalatorLabel(): String? {
        return escalatorLabel
    }

    fun escalator(): String? {
        return escalator
    }

    fun escalatorValue(): String? {
        return escalatorValue
    }

    fun escalatorDefault(): Int {
        return escalatorDefault
    }

    fun escalatorMax(): Int {
        return escalatorMax
    }

    fun escalatorStep(): Int {
        return escalatorStep
    }

    fun resetButton(): String? {
        return resetButton
    }

    fun submitButton(): String? {
        return submitButton
    }

    fun suggestedContractsTable(): String? {
        return suggestedContractsTable
    }

    fun suggestedContractsMainHeader(): String? {
        return suggestedContractsMainHeader
    }

    fun suggestedContractsSubHeaderYears(): String? {
        return suggestedContractsSubHeaderYears
    }

    fun suggestedContractsSubHeaderSalary(): String? {
        return suggestedContractsSubHeaderSalary
    }

    fun suggestedContractsSubHeaderRemuneration(): String? {
        return suggestedContractsSubHeaderRemuneration
    }

    fun suggestedContractsSubHeaderContractYear(): String? {
        return suggestedContractsSubHeaderContractYear
    }

    fun suggestedContractsSubHeaderContractYearSalary(): String? {
        return suggestedContractsSubHeaderContractYearSalary
    }

    fun suggestedContractsSubHeaderContractYearlyBonus(): String? {
        return suggestedContractsSubHeaderContractYearlyBonus
    }

    fun suggestedContractsSubHeaderContractYearlyTotal(): String? {
        return suggestedContractsSubHeaderContractYearlyTotal
    }

    fun suggestedContractsContractYears(): String? {
        return suggestedContractsContractYears
    }

    fun suggestedContractsTotalSalary(): String? {
        return suggestedContractsTotalSalary
    }

    fun suggestedContractsTotalRemuneration(): String? {
        return suggestedContractsTotalRemuneration
    }

    fun suggestedContractsCurrentYear(): String? {
        return suggestedContractsCurrentYear
    }

    fun suggestedContractsYearSalary(): String? {
        return suggestedContractsYearSalary
    }

    fun suggestedContractsBonusPerYear(): String? {
        return suggestedContractsBonusPerYear
    }

    fun suggestedContractsTotalPerYear(): String? {
        return suggestedContractsTotalPerYear
    }
}
